### PR TITLE
Add proof possession revocation for PKI secrets engine

### PR DESCRIPTION
**Pull request by cipherboy**
This PR adds support to the PKI secrets engine for revocation of certs without requiring privileged access to the `/revoke` endpoint. In particular, we use the private key as evidence of permission to revoke the certificate: either the private key was leaked, therefore suggesting the certificate should be revoked anyways, or someone with access to the private key material (and thus, likely the original issuer) suggests the certificate should be revoked and could publish the private key publicly (therefore, triggering case one).

This is done by hitting a new `/revoke-with-key` endpoint with the `private_key` field (taking a PEM-encoded private key). This endpoint otherwise has semantics similar to `/revoke` (allowing it to either revoke by BYOC or stored cert serial number), but is less privileged.

---

- [x] #16564 to update and merge
- [x] Tests for PoP +/- BYOC revocation
- [x] Change log entry
- [x] Finish discussion on whether or not it should be authed or unauthed. 
- [x] Documentation

### CODE REVIEW

**Comment by cipherboy** [comment]
@kitography / @stevendpclark -- this is ready for review now \o/

**Review by stevendpclark — APPROVED** [review]
👍 one small nit of an additional test request.

File: builtin/logical/pki/crl_test.go [hunk 1]
```diff
@@ -420,6 +420,148 @@ func TestBYOC(t *testing.T) {
...
+		"pem_bundle": oldRoot,
+	})
+	require.NoError(t, err)
+
+	_, err = CBWrite(b, s, "revoke-with-key", map[string]interface{}{
+		"certificate": notStoredCert,
+		"private_key": notStoredKey,
+	})
+	require.NoError(t, err)
+}
```
[the remark this answers is not in the document]
Comment (cipherboy) [reply builtin/logical/pki/crl_test.go:563]: Added, lmk if you like that :-)

[the remark this answers is not in the document]
Comment (stevendpclark) [reply builtin/logical/pki/crl_test.go:579]: Thanks much appreciated!

**Review by stevendpclark — APPROVED** [review]

**Pull request merged**